### ISSUE TITLE
Bug: New Columns Not Appearing in Alpha Version

### ISSUE THREAD

**Issue by cristobalsalcedo90** [deneb-viz/deneb#367]
Daniel Marsh-Patrick first and foremost, I want to express my sincere gratitude for your excellent work in developing Deneb. Your contribution to Power BI is truly valuable and highly appreciated by the community.

Now, I'd like to share some observations I've made while testing the alpha version of Deneb. I downloaded the alpha version and was testing it out. I noticed a peculiar behavior when using it for the first time in a new Power BI project. While working with Vega and making some transformations that added new columns, they didn't appear immediately, not even after closing and reopening Deneb. I conducted several tests, and the issue persisted. However, when I switched back to Deneb version 1.5, the new columns appeared as expected.

Subsequently, I returned to the alpha version of Deneb, and this time the new columns did appear as intended. I've attached a .gif image that demonstrates this peculiar behavior. I'm not sure if this might be a problem, but I wanted to bring it to your attention. Thank you for your assistance!

![image](https://github.com/cristobalsalcedo90/cristobalsalcedo.github.io/blob/16c55d3635661256feb3386417c03792761682f4/4CJOoIRten20230915-122504%20Untitled_-_Power_BI_Desktop.gif?raw=true)

**Comment by dm-p** [comment]
This looks to be an issue with (a) Vega and (b) changes to the root dataset not being picked up as volatile for the debug viewer.

**Comment by dm-p** [comment]
Hey, @cristobalsalcedo90: can I confirm if you're using the latest alpha build, or one from at least this week? Just want to be sure, as I can't seem to reproduce. Steps I took are as follows:

1. Add new visual to canvas (Deneb 1.6 alpha).
2. Add data to visual.
3. Open editor.
4. Select Vega as language and simple bar chart as template.
5. Assign column and measure.
6. Add transform to `dataset`:
````json
{
  "data": [
    {
      "name": "dataset",
      "transform": [
        {
          "type": "formula",
          "expr": "null",
          "as": "__x__"
        }
      ]
    }
  ],
  ...
}
````

For me, the `__x__` column is present in the debug pane and have attached a video confirming this behavior:

https://github.com/deneb-viz/deneb/assets/10572054/a2953aa6-9076-4f58-b3ff-55a914cae711

----

I may have found a different bug while investigating, so I'll work on that one (in case it's related), but it would be good to know any further details that will help differentiate, such as:

- Version of Power BI Desktop you're running.
- Whether you have any specific preview features enabled or disabled.

If you haven't tried the latest alpha build (1.6.0.20230915.549b95d), it would be good if you could download and double-check there also. If you can reproduce, any further steps that can help me reproduce on my end will be greatly appreciated :)

**Comment by dm-p** [comment]
I have raised #369 for the other issue I found while investigating this one, so that they can be tracked. They may be related, but we'll see how things go.

**Comment by cristobalsalcedo90** [comment]
Hi Daniel,

I appreciate your prompt response in addressing the issue with the alpha
version.

I wanted to inform you that the problem seems to be related to an earlier
version of Deneb
(ALPHAdeneb7E15AEF80B9E4D4F8E12924291ECE89A.1.6.0.20230907.f92a646.pbiviz).

I downloaded the version you mentioned later (1.6.0.20230915.549b95d), and
the issue is no longer present. However, during my testing, I noticed
something related to the other problem you mentioned. It appears that when
I create a new column and subsequently remove that specification, the
information remains saved even after closing and reopening Deneb. It only
gets cleared if I switch to another page in the report and then return.

My Power BI Desktop is on the August 2023 version. Additionally, I've
created a GIF video that demonstrates this behavior, and I'd like to attach
the PBIX file for your reference.

Thank you once again for your assistance in addressing these issues.


Link to the GIF Video:

https://github.com/cristobalsalcedo90/cristobalsalcedo.github.io/blob/6f056d759019f178e7bd07ecf32f60581e08d88c/TEST%20Version%20(ALPHAdeneb7E15AEF80B9E4D4F8E12924291ECE89A.1.6.0.20230915.549b95d).gif

PBIX
https://github.com/cristobalsalcedo90/cristobalsalcedo.github.io/raw/main/TEST%20Version%20(ALPHAdeneb7E15AEF80B9E4D4F8E12924291ECE89A.1.6.0.20230915.549b95d).pbix
